src/vorta/views/repo_add_dialog.py

```diff
@@ -234,12 +262,41 @@ def __init__(self):
         self.repoDataFormLayout.addRow(self.passwordLabel, self.passwordInput)
 
     def set_password(self, URL):
-        '''Autofill password from keyring only if current entry is empty'''
+        """Autofill password from keyring only if current entry is empty"""
         password = VortaKeyring.get_keyring().get_password('vorta-repo', URL)
         if password and self.passwordInput.get_password() == "":
             self._set_status(self.tr("Autofilled password from password manager."))
             self.passwordInput.setText(password)
 
+    def run_result(self, result):
+        """Handle result of BorgInfoRepoJob."""
+        self.saveButton.setEnabled(True)
+        if result['returncode'] == 0:
+            self.added_repo.emit(result)
+            self.accept()
+        else:
+            self._set_status(self.tr('Unable to add your repository.'))
+            for error in result['errors']:
+                if 'msgid' in error and error['msgid'] in ['Repository.DoesNotExist', 'Repository.InvalidRepository']:
+                    self.init_new_repo()
+                    break
```

Comment:
Not sure. E.g. by using the cmd, can you find out what the command outputs as an error/warning. Use the json flag so that you can find out which `msgid` is used.

Reply:
Got it, for remote repository it was just `InvalidRepository`
So now it works for both local and remote repositories.